assessments/projects/cni/self-assessment.md

```diff
@@ -0,0 +1,277 @@
+
+### Actions
+
+#### Action Overview
+
+1. A format for administrators defines network configurations
+2. A protocol for container runtimes makes requests to network plugins with parameters from environment variables and network configuration
+3. With the supplied configuration, a procedure executes the plugins.
+4. CNI plugins may also have a procedure to delegate functionality to other plugins
+5. Data types for plugins return their results to the runtime
+
+#### Action 1: Network Configuration Format
+
+CNI defines a network configuration format for administrators. It contains directives for both the container runtime as well as the plugins
+to consume. At plugin execution time, this configuration format is interpreted by the runtime and transformed into a form to be passed to the plugins.
+* **Example configuration**:
+  <br>
+  <div style="text-align:center;">
+    <img src="docs/CNI-config.png" alt="CNI config">
+  </div>
+  <br>
+
+#### Action 2: Execution Protocol
+
+The CNI protocol is based on execution of binaries invoked by the container runtime. CNI defines the protocol between the plugin binary and the runtime. CNI defines 4 operations: ADD, DEL, CHECK, and VERSION. These are passed to the plugin via the CNI\_COMMAND environment variable.
+
+#### Action 3: Execution of Network Configurations
+
+A container time interprets a network configuration and executes plugins accordingly. A runtime can add, delete, or check a network configuration in a container, which results in a series of plugin ADD, DELETE, or CHECK executions correspondingly.
+
+#### Action 4: Plugin Delegation
+
+There are some operations that, for whatever reason, cannot reasonably be implemented as a discrete chained plugin. Rather, a CNI plugin may wish to delegate some functionality to another plugin. One common example of this is IP address management.
+
+#### Action 5: Result Types
+
+Plugins can return one of three result types:
+* Success
+* Error
+* \_Version
+
+### Goals
+
+**General**
+* CNI defines a common interface between the network plugins and container execution
+* CNI is language-agnostic and a vendor-neutral specification
+* Backwards compatible: plugins are able to easily implement all versions of the specification and some helper code is available to convert between versions.
+
+**Security**
+* Limit which file paths are included when searching for a CNI plugin on a system.
+
+### Non-goals
+
+**General**
+* Dynamic updates to existing network configurations
+* Dynamic policies for network bandwidth and firewall rules
+
+**Security**
+* Stop a container runtime from executing CNI plugins
+* Restrict what a plugin does
+* Manage the security of the container it interacts with
+
+## Self-assessment use
+
+This self-assessment is created by the CNI team to perform an internal analysis of the
```

Comment:
Was this created with active participation and review by the CNI team? If not, please remove that reference or replace it with a more accurate statement.

Consider listing the assessment contributors in the topmost paragraph, below the title and before the Table of Contents.